db/models/models.js

```diff
@@ -6,26 +6,128 @@ exports.locateTopics = (stuff) => {
   });
 };
 
-exports.locateArticleId = (article_Id) => {
-  if (article_Id > 500) {
+exports.locateArticleId = async (article_Id) => {
+  const queryArticles = db.query(
+    `SELECT * FROM articles WHERE article_id = ${article_Id};`
+  );
+  const result = await queryArticles;
+
+  if (result.rows.length === 0) {
     return Promise.reject({ status: 404, msg: "not found" });
   }
 
-  console.log(article_Id, "fail test");
-  let querys = `SELECT * FROM articles WHERE article_id = ${article_Id};`;
-  console.log(querys);
-  return db.query(querys).then((results) => {
-    return results.rows[0];
-  });
+  const queryComments = db.query(
+    `select * from comments where author = '${result.rows[0].author}';`
+  );
+  const newResult = await queryComments;
+
+  result.rows[0]["comment_count"] = newResult.rows.length;
+
+  return result.rows[0];
 };
 
 exports.updateArticleId = async (article_id, newVote) => {
-  //   console.log(newVote, "vote");
   const query = `UPDATE articles SET votes = votes + ${newVote} WHERE article_id = ${article_id} RETURNING *;`;
-  //   console.log(query);
+
+  const result = await db.query(query);
+  if (result.rows.length === 0) {
+    return Promise.reject({ msg: "not found", status: 404 });
+  }
+  return result.rows[0];
+};
+
+exports.locateUsers = async () => {
+  const query = `SELECT * FROM users`;
   const result = await db.query(query);
+  return result.rows;
+};
+
+exports.locateArticles = async (order_by) => {
+  const articleColumns = [
+    "title",
+    "topic",
+    "author",
+    "body",
+    "created_at",
+    "votes",
+  ];
+  const articleQuerys = ["order", []];
+
+  let querys = `SELECT * FROM articles`;
+  let arrayForValue = [];
+  const keyCheckOrderBy = Object.keys(order_by);
+  Object.keys(order_by).forEach((value) => arrayForValue.push(order_by[value]));
+  const isEmpty = (order_by) => {
+    return Object.keys(order_by).length !== 0;
+  };
+  if (
+    !articleQuerys.includes(keyCheckOrderBy[0]) &&
+    keyCheckOrderBy.length !== 0
+  ) {
+    return Promise.reject({ status: 400, msg: "invalid request" });
+  }
+  if (!articleColumns.includes(keyCheckOrderBy[0])) {
+    if (isEmpty(order_by)) querys += ` ORDER BY ${order_by.order}`;
+  }
+  if (articleColumns.includes(keyCheckOrderBy[0])) {
+    querys += ` WHERE ${keyCheckOrderBy[0]} = '${arrayForValue[0]}'`;
+  }
```

Comment:
There's quite a lot going on here - I think some of it will be able to be streamlined somewhat by not having to worry about handling non-existent queries as mentioned above, i.e. `?hello=shaun` should be ignored entirely.